v2/rust/src/setup.rs

```diff
@@ -17,7 +20,9 @@ pub fn setup(config: &Config) -> Result<()> {
     create_dir_all(&config.results_directory).context("Failed to create results directory")?;
     create_dir_all(suite_results_directory(&config.results_directory))
         .context("Failed to create suite results directory")?;
-    clean_up_results_directory_atomic(config)
+    clean_up_results_directory_atomic(config)?;
+    adjust_user_permissions(config);
```

Comment:
Should we not pass these errors?

Reply:
We have two options here:
1. Pass the error. Since we have no good way of handling the error, our setup would simply stop and pass the error to the main function.
2. Log the error and continue with the next suite.

I vote for option 2 (of course Simon should ultimately decide this). It's true that all other steps above immediately lead to an abort. However, errors in these steps indicate I/O issues, which are quite severe, so I think an abort is justified here. On the other hand, I am not sure how severe an error in the permissions step is.

Reply:
Hmm, seems like we should maybe at least forward this error to Checkmk?

Reply:
After talking to Simon, we decided to do this a bit differently: Instead of setting permissions for individual users, we set permissions for the group `Users`. And if this fails, the scheduler is allowed to crash.